### ISSUE TITLE
support wired networks

### ISSUE THREAD

**Issue by CWempe** [marvinroger/homie-esp8266#31]
It would be great if Homie would support wired networks, too.

Like the ENC28J60 chip.
https://www.microchip.com/wwwproducts/en/en022889

This might be usefull for cases where the wifi signal is to weak.

**Comment by marvinroger** [comment]
Thinking about it, how would we implement the config mode? Would it even exist?

**Comment by CWempe** [comment]
There are my thought on this topic.
Just off the top of my head.

Since you usually don't need authentication for wired networks the device should connect to the existing network without problems and get an IP via DHCP.
The (Web)App could discover the device via mDNS or the user could enter the IP.

The user could configure the parameters via Serial.

Optionaly hardcode the parameters in the sketch.

The device could have an initiating ip address (like 10.1.1.1) and the user could connect by configure a corosponding ip on his computer. (not the easiest way)

**Comment by marvinroger** [comment]
> The (Web)App could discover the device via mDNS or the user could enter the IP.

For the mDNS part, no it can't, a web app can only communicate through HTTP. This would require native code. But implementing mDNS would be an interesting idea for 3rd party configurators using native code. The user could also enter the IP, that's right.

> Optionaly hardcode the parameters in the sketch.

What is good about Homie sketches is that they are portable. There are no hardcoded credentials, so you can share the firmwares. I don't want to break this flexibility.

> The user could configure the parameters via Serial.

This is already possible, you can upload `/homie/config.json` to the SPIFFS.

I am not sure when I will implement this, if I ever. This is a big refactoring and even the name would need to change (Homie for ESP8266 -> Homie for Arduino).

**Comment by CWempe** [comment]
To be clear: I am not planing to use wired devices.

It's just a (valid) idea that came up, when I discussed it with a friend who has many ethernet cables in his house. ;)

**Comment by marvinroger** [comment]
I tried to picture how I could implement this, and the conclusion I came to was that it is not feasible in the current form of the project. On wired devices (in other words, genuine Arduino), there is no SPIFFS interface, no SSL support, and most importantly not enough RAM. The project would need to be rebuilt from the ground up, and I don't have enough resources to. This might be a good idea for a fork, though!
